pkg/didcomm/transport/http/outbound_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+	Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+	SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+const certPrefix = "../../../../test/fixtures/keys/"
+
+func TestWithOutboundOpts(t *testing.T) {
+	opt := WithOutboundCommHTTP(nil)
+	require.NotNil(t, opt)
+	clOpts := &outboundCommHTTPOpts{}
+	opt(clOpts)
+
+	opt = WithOutboundTimeout(10 * time.Second)
+	require.NotNil(t, opt)
+	clOpts = &outboundCommHTTPOpts{}
+	// opt.client is nil, so setting timeout should panic
+	require.Panics(t, func() { opt(clOpts) })
+
+	opt = WithOutboundTLSConfig(nil)
+	require.NotNil(t, opt)
+	clOpts = &outboundCommHTTPOpts{}
+	opt(clOpts)
+}
+
+func TestOutboundHTTPTransport(t *testing.T) {
+	// prepare http server
+	server := startMockServer()
+	defer func() {
+		err := server.Close()
+		if err != nil {
+			log.Fatalf("Failed to stop server: %s", err)
+		}
+	}()
+	//build a mock cert pool
+	cp := x509.NewCertPool()
+	err := addCertsToCertPool(cp)
+	require.NoError(t, err)
+
+	// build a tls.Config instance to be used by the outbound transport
+	tlsConfig := &tls.Config{
+		RootCAs:      cp,
+		Certificates: nil,
+	}
+	// create a new invalid Outbound transport instance
+	ot, err := NewOutbound()
+	require.Error(t, err)
+	require.EqualError(t, err, "Can't create an outbound transport without an HTTP client or tls.Config option")
+
+	// now create a new valid Outbound transport instance and test its Send() call
+	ot, err = NewOutbound(WithOutboundTLSConfig(tlsConfig), WithOutboundTimeout(1*time.Second))
+	require.NoError(t, err)
+	require.NotNil(t, ot)
+
+	// test mock HTTPHandler
+	req, err := http.NewRequest("POST", "https://localhost:8090", nil)
```

Comment:
All "valid" instance of port 8090 should be replaced with the port returned by spinning up http server.

Reply:
done